src/linear_eq.jl

```diff
@@ -48,119 +46,103 @@ function gauss_seidel_interval!(x::AbstractArray, A::AbstractMatrix, b::Abstract
     x
 end
 
-function gauss_seidel_contractor(A::AbstractMatrix, b::AbstractArray; precondition=true, maxiter=100)
+function gauss_seidel_contractor(A::SMatrix{N, N, Interval{T}},
+        b::SVector{N, Interval{T}} ;
+        precondition=true, maxiter=100) where {N, T}
+
+    x = MVector{N, Interval{T}}([entireinterval(T) for _ in 1:N])
+    Am = convert(MMatrix{N, N, Interval{T}}, A)
+    bm = convert(MVector{N, Interval{T}}, b)
+
+    gauss_seidel_contractor!(x, Am, bm, precondition=precondition, maxiter=maxiter)
 
-    n = size(A, 1)
-    x = similar(b)
-    x .= -1e16..1e16
-    x = gauss_seidel_contractor!(x, A, b, precondition=precondition, maxiter=maxiter)
     return x
 end
 
-function gauss_seidel_contractor!(x::AbstractArray, A::AbstractMatrix, b::AbstractArray; precondition=true, maxiter=100)
+function gauss_seidel_contractor!(x::MVector{N, Interval{T}},
+        A::MMatrix{N, N, Interval{T}},
+        b::MVector{N, Interval{T}} ;
+        precondition=true, maxiter=100) where {N, T}
 
     precondition && ((A, b) = preconditioner(A, b))
 
-    n = size(A, 1)
-
-    diagA = Diagonal(A)
+    diagA = diag(A)
     extdiagA = copy(A)
-    for i in 1:n
-        if (typeof(b) <: SArray)
-            extdiagA = setindex(extdiagA, Interval(0), i, i)
-        else
-            extdiagA[i, i] = Interval(0)
-        end
+    for i in 1:N
+        extdiagA[i, i] = Interval(zero(T))
     end
-    inv_diagA = inv(diagA)
+    inv_diagA = 1 ./ diagA
 
     for iter in 1:maxiter
```

Comment:
Maybe there should be a more sophisticated convergence check here that checks some kind of rate of convergence?

Reply:
The reference given (Eldon Hansen and G. William Walster : Global Optimization Using Interval Analysis - Chapter 5 - Page 115) does only state the process can be iterated until no changes happen. I'll have to dig a bit more in the litterature to see if something more subtle exist.

Reading the litterature in more detail may also indicate if Gauss elimination is the best choice as default linear equation solver.